core/src/oauth/connection.rs

```diff
@@ -21,6 +21,26 @@ use tracing::error;
 
 use super::providers::confluence::ConfluenceConnectionProvider;
 
+// We hold the lock for at most 15s. In case of panic preventing the lock from being released, this
+// is the maximum time the lock will be held.
+static REDIS_LOCK_TTL_SECONDS: u64 = 15;
+// To ensure we don't write without holding the lock providers must comply to this timeout when
+// operating on tokens.
+pub static PROVIDER_TIMEOUT_SECONDS: u64 = 10;
+
+lazy_static! {
```

Comment:
any specific reason to have 2 distinct lazy static blocks ?

Reply:
nope will fix